pkg/k8shandler/rbac.go

```diff
@@ -154,6 +154,19 @@ func createOrUpdateClusterRoleBinding(roleBinding *rbac.ClusterRoleBinding) erro
 	return nil
 }
 
+func combineSubjects(a, b []rbac.Subject) []rbac.Subject {
```

Comment:
To address @ewolinetz concerns.  Consier treating this as sets and handling in that fashion by merging like keys.  @ewolinetz would that resolve your concerns?

Reply:
slightly, we keep appending every time there's a new change... its not just repeating keys that concern me. We simply say "take our current keys and append these specified ones" this sort of goes against the notion that we have a single source of truth for what our objects should look like.

Reply:
The other approach is to make hardcoded cluster roles  `elasticsearch-proxy`, `elasticsearch-metrics` and create separate instances per ES instance `<es-name>-proxy|metrics`. WDYT? 

I am not sure if it would produce any breaking changes on your side?

Reply:
I have updated the PR. Now the subject list is constructed in the operator by querying all ES instances - as we discussed on slack.